### ISSUE TITLE
bug in addConsSOS2

### ISSUE THREAD

**Issue by mueldgog** [SCIP-Interfaces/PySCIPOpt#259]
Using the debug library of the SCIP Optimization Suite (version 6.0) together with this minimal example

```
>>> from pyscipopt import Model
>>> m = Model()
>>> vars = [m.addVar() for i in range(5)]
>>> m.addConsSOS2(vars)
```

results in the following error

```
scip/src/scip/cons_sos2.c:456: appendVarSOS2: Assertion `consdata->weights != NULL || consdata->nvars > 0' failed.
```

**Comment by mattmilten** [comment]
Strange. When I run the code in debug mode I get a different error:

```
[/scipoptsuite/scip/src/blockmemshell/memory.c:2432] ERROR: 120 bytes (1 elements of size 120) not freed. First Allocator: /mnt/c/Users/m2/ZIB/scipoptsuite/scip/src/scip/cons_soc.c:5095
[/scipoptsuite/scip/src/blockmemshell/memory.c:2432] ERROR: 64 bytes (1 elements of size 64) not freed. First Allocator: /mnt/c/Users/m2/ZIB/scipoptsuite/scip/src/scip/cons_nonlinear.c:9300
[/scipoptsuite/scip/src/blockmemshell/memory.c:2432] ERROR: 40 bytes (1 elements of size 40) not freed. First Allocator: /mnt/c/Users/m2/ZIB/scipoptsuite/scip/src/scip/misc.c:2903
[/scipoptsuite/scip/src/blockmemshell/memory.c:2432] ERROR: 16 bytes (1 elements of size 16) not freed. First Allocator: /mnt/c/Users/m2/ZIB/scipoptsuite/scip/src/nlpi/expr.c:12076
[/scipoptsuite/scip/src/blockmemshell/memory.c:2450] ERROR: 240 bytes not freed in total.
```

**Comment by mattmilten** [comment]
And when I use SCIP without block- or bufmem then there is no error at all.
Can you confirm that you are using SCIP 6.0.1?

**Comment by mueldgog** [comment]
The issue has been fixed with 6.0.1. I have seen that Marc fixed the issue in cons_sos2.c
